helix-core/src/main/java/org/apache/helix/model/builder/IdealStateBuilder.java

```diff
@@ -85,12 +87,13 @@
    * A constraint that limits the maximum number of partitions per Node.
    */
   private int maxPartitionsPerNode;
+
   /**
    * Allocate the resource to nodes that are tagged with a specific "nodeGroup"
    * name. By default a resource will be allocated to all nodes registered to
    * the cluster.
    */
-  private String nodeGroup = "*";
+  private String nodeGroup = null;
```

Comment:
Does it still apply to the java doc "By default a resource will be allocated to all nodes" - do we need to change the java doc?

Reply:
The original code is not fitting the comment. The reason why it is not failing is that the value is not passed.

Please also help to double-check the code. I think "*" will cause the resource to be assigned onto the node with "*" tag. I don't see any wildcard fitting logic in the rebalancer. null is the safest option.